livekit-android-sdk/src/main/java/io/livekit/android/room/RTCEngine.kt

```diff
@@ -48,22 +46,25 @@ internal constructor(
             when (value) {
                 IceState.CONNECTED -> {
                     if (oldVal == IceState.DISCONNECTED) {
-                        LKLog.d { "publisher ICE connected" }
+                        LKLog.d { "primary ICE connected" }
                         listener?.onIceConnected()
                     } else if (oldVal == IceState.RECONNECTING) {
-                        LKLog.d { "publisher ICE reconnected" }
+                        LKLog.d { "primary ICE reconnected" }
                         listener?.onIceReconnected()
                     }
                 }
                 IceState.DISCONNECTED -> {
-                    LKLog.d { "publisher ICE disconnected" }
-                    listener?.onDisconnect("Peer connection disconnected")
+                    LKLog.d { "primary ICE disconnected" }
```

Comment:
We were debating this last week. `DISCONNECTED` is technically a temporary disconnection and ICE can connect back up. That happens at 5 seconds after media stops in Chrome (not sure about other browsers or mobile). `FAILED` means the ICE agent has given up and we can try reconnecting.

I think we decided we should act on `FAILED` initially and maybe think about doing a reconnect on `DISCONNECTED` (or some other rule) once we get some soak in the reconnect scenarios.

I maybe remembering things wrong. Just wanted to confirm.

Reply:
does this means if pc.IceConnectionState is DISCONNECTED or FAILED, we will do reconnect?